lib/posix/pthread_rwlock.c

```diff
@@ -231,17 +242,32 @@ static u32_t write_lock_acquire(pthread_rwlock_t *rwlock, s32_t timeout)
 {
 	u32_t ret = 0U;
 	s64_t elapsed_time, st_time = k_uptime_get();
+	k_timeout_t k_timeout;
+
+	/* Convert integer timeout to k_timeout_t. Code structured in a way
+	 * that a smart compiler may optimize it out (for a common definition
+	 * of K_FOREVER.
+	 */
+	if (timeout == -1) {
+		k_timeout = K_FOREVER;
+	} else {
+		k_timeout = K_MSEC(timeout);
+	}
 
 	/* waiting for release of write lock */
-	if (k_sem_take(&rwlock->wr_sem, timeout) == 0) {
-		if (timeout > K_NO_WAIT) {
+	if (k_sem_take(&rwlock->wr_sem, k_timeout) == 0) {
+		/* update remaining timeout time for 2nd sem */
+		if (timeout == -1) {
```

Comment:
Consider using `SYS_TIMEOUT_MS` and in any case use `SYS_FOREVER_MS` to compare against